### PR TITLE
Enhancements to the Fixed type

### PR DESCRIPTION
**Pull request by dfrg**
I've run into some friction when using the fixed point types for computing things like deltas and variation region scalars. This adds `Mul`, `Div` and `Neg` impls for `Fixed` along with exposed associated constants for 0 and 1. Also adds a `to_fixed` conversion for `F2dot14`.

Ok to merge if there are no issues.

### CODE REVIEW

File: font-types/src/fixed.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //! fixed-point numerical types
 
-use std::ops::{Add, AddAssign, Sub, SubAssign};
+use core::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign};
```
Comment (cmyr) [inline font-types/src/fixed.rs:3]: I have `core` aliased as `std` if we're `no_std`. I'm not sure where I got this pattern from, but I don't think I invented it. In any case, I think we use `std` here and if we want to change the overall pattern we can do that in a separate PR?

**Review by cmyr — APPROVED** [review]
Looks good. These are definitely still rough around the edges; in particular I don't like the `from_` methods (`from_f64` here, and `from_f32` on `F2Dot14`) and think they should both maybe just have `fn new` constructors? But that has no impact here.


This is also the kind of code that I would generally write unit tests for, but then it's also some of the code I am least confident writing so ymmv.

File: font-types/src/fixed.rs (hunk 1 above)
Comment (dfrg) [reply font-types/src/fixed.rs:3]: oh! I actually thought I messed this up by doing automatic imports so this was my attempt to fix it. I have no objections to that pattern so I'll revert this change.

**Comment by dfrg** [comment]
I added a couple unit tests for mul/div. I also find the constructors a bit less than satisfying and I honestly would like access to the inner values without moving through from/to_be_bytes(), but I'm trying to resist and see how far I can get without them. We'll revisit as things progress.

**Comment by cmyr** [comment]
If you want access to the inner values I think a `to_bits` and `from_bits` method would be fine? These exist on `f32` and `f64`, so there's precedent.

**Comment by cmyr** [comment]
But I guess I'd also be curious where you need this, it does feel like an abstraction leak?
